clang/lib/Driver/Driver.cpp

```diff
@@ -4791,6 +4791,20 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
     if (!UnbundlerInputs.empty()) {
       Action *PartialLink =
           C.MakeAction<PartialLinkJobAction>(UnbundlerInputs, types::TY_Object);
+      if (!LastArg) {
```

Comment:
Is there an example of such an occurrence? I.e. `LastArg` being set to `nullptr` in scope of `addUnbundlerInput()`. It's just that I'm personally stuck with 100% mapping the new piece of code onto the flow.

Reply:
Yes, following test populates ```UnbundlerInputs``` yet ```LastArg``` is ```null```.
   
``` clangxx -target x86_64-unknown-linux-gnu -fsycl -L/dummy/dir %t.o -Wl,--whole-archive %t.a %t_2.a -Wl,--no-whole-archive -### ```

Here is what we are doing:

if ```LastArg``` is ```null```, we set it to last ```HostAction```'s ```Arg```. But we will have complexity when we encounter an ```offloadAction``` for which we are setting it to corresponding ```HostAction```

Reply:
Thanks, it's clearer now!